tables/conditions.py

```diff
@@ -253,8 +253,13 @@ def fix_invert(idxcmp, exprnode, indexedcols):
             return [expr]
 
     # Recursively get the expressions at the left and the right
-    lexpr = _get_idx_expr_recurse(left, indexedcols, idxexprs, strexpr)
+    lexpr = deepcopy(_get_idx_expr_recurse(left, indexedcols, idxexprs,
+                                           strexpr))
+    if len(lexpr) == 2 and lexpr != not_indexable:
```

Comment:
This is a bit of a hack. This is now necessary since `strexpr` is no longer a (reference to) a variable in a high scope (a sort of global).
It can probably be fixed by refactoring `add_expr` below.